lib/sensors/gyroscope.cpp

```diff
@@ -0,0 +1,122 @@
+#include "gyroscope.hpp"
+
+namespace hyped::sensors {
+
+Gyroscope::Gyroscope(core::ILogger &logger, io::II2c &i2c, const std::uint8_t channel)
+    : logger_(logger),
+      i2c_(i2c),
+      channel_(channel)
+{
+}
+
+Gyroscope::~Gyroscope()
+{
+}
+
+const std::optional<std::int16_t> Gyroscope::read(core::Axis axis)
+{
+  switch (axis) {
+    case core::Axis::kX: {
+      const auto gyroscope_x_high_byte = i2c_.readByte(kDefaultGyroscopeAddress, kDataXHigh);
+      if (!gyroscope_x_high_byte) {
+        logger_.log(
+          core::LogLevel::kFatal, "Failed to read gyroscope X-axis high at channel %d", channel_);
```

Comment:
I thought the gyroscope only mattered if the active suspension was being used? Therefore would it not make more sense to log fatal only if active suspension is being used and just logging `kWarn` if active suspension is disabled @SnickeyX @kshxtij ??

Reply:
This just doesn't seem like it would always be worth failing the run over...

Reply:
thats an amazing catch, going to push for my overhaul to get merged to we can merge master and get `kWarn` here